python/cuml/dask/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,205 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+
+
+from uuid import uuid1
+
+from cuml.naive_bayes import MultinomialNB as MNB
+
+import dask
+
+from cuml.dask.common import extract_arr_partitions, \
+    workers_to_parts
+
+from dask.distributed import default_client
+
+
+class MultinomialNB(object):
+
+    """
+    Distributed Naive Bayes classifier for multinomial models
+
+    The multinomial Naive Bayes classifier is suitable for classification
+    with discrete features (e.g., word counts for text classification).
+
+    The multinomial distribution normally requires integer feature counts.
+    However, in practice, fractional counts such as tf-idf may also work.
+    """
+    def __init__(self, client=None, **kwargs):
+
+        """
+        Create new multinomial distributed Naive Bayes classifier instance
+
+        Parameters
+        -----------
+
+        client : dask.distributed.Client optional Dask client to use
+        """
+
+        self.client_ = client if client is not None else default_client()
+        self.model_ = None
+        self.kwargs = kwargs
+
+    @staticmethod
+    def _fit(Xy, classes, kwargs):
+
+        model = MNB(**kwargs)
+
+        for x, y in Xy:
+            model.partial_fit(x, y, classes=classes)
+        return model.class_count_, model.feature_count_
+
+    @staticmethod
+    def _predict(model, X):
+        return [model.predict(x) for x in X]
+
+    def fit(self, X, y, classes=None):
+
+        """
+        Fit distributed Naive Bayes classifier model
+
+        Parameters
+        ----------
+
+        X : dask.Array with blocks containing dense or sparse cupy arrays
+        y : dask.Array with blocks containing cupy.ndarray
+        classes : array-like containing unique class labels
+
+        Returns
+        -------
+
+        cuml.dask.naive_bayes.MultinomialNB current model instance
+        """
+
+        # Only Dask.Array supported for now
+        if not isinstance(X, dask.array.core.Array):
+            raise ValueError("Only dask.Array is supported for X")
+
+        if not isinstance(y, dask.array.core.Array):
+            raise ValueError("Only dask.Array is supported for y")
+
+        if len(X.chunks[1]) != 1:
+            raise ValueError("X must be chunked by row only. "
+                             "Multi-dimensional chunking is not supported")
+
+        worker_parts = self.client_.sync(extract_arr_partitions,
+                                         [X, y])
+
+        worker_parts = workers_to_parts(worker_parts)
+
+        n_features = X.shape[1]
+
+        classes = cp.unique(y.map_blocks(cp.unique).compute()) \
```

Comment:
It would probably make sense to use it for the local ones on the workers too, right? Do you know if the serialization (and UCX support) supporting RMM allocated memory yet?